bundles/org.openhab.binding.modbus.solaxx3mic/README.md

```diff
@@ -0,0 +1,296 @@
+Java code is written that way, that is scans all defined and mapped to items channels and maps received modbus registers block to them dynamically.
+
+## Discovery
+
+Because of flexibility of this binding discovery is not supported.
+
+## Binding Configuration
+
+Binding does not requires any special configuration.
+
+## Thing Configuration
+
+Thing configuration is necessary in configuration file because of extended amount of information required by thing.
+
+Basically this thing requires Bridge of type Modbus TCP or Modbus Serial.
+It also requires definition of following parameters:
+
+1. refresh - amount of time between polls [s]
+2. inputAddress - begin of modbus type 3 registers block
+3. inputBlockLength - length of modbus type 3 registers block
+4. maxTries - maximal amount of tries during failure condition
+5. holdingAddress - begin of modbus type 4 registers block
+6. holdingBlockLength - length of modbus type 4 registers block
+
+
+## Channels
+
+Channels are splitted into several groups.
+All channels are read only.
+Not all channels are required to use.
+
+Groups:
+
+1. Device information
+2. AC Line General
+3. AC Phases A, B, C as separate Groups
+4. DC Lines 1 and 2 as separate Groups
+
+Channels in every group type:
+
+1. Device Information (device-information)
+   - status (status-type)
+   - cabinet-temperature (cabinet-temperature-type)
+   - heatsink-temperature(heatsink-temperature-type)
+2. AC Summary (ac-general)
+   - ac-power (ac-power-type)
+   - ac-dayly-energy (ac-dayly-energy-type)
+   - ac-lifetime-energy (ac-lifetime-energy-type)
+3. AC Phase (ac-phase)
+   - ac-phase-voltage-to-n (ac-phase-voltage-to-n-type)
+   - ac-phase-current (ac-phase-current-type)
+   - ac-phase-frequency (ac-phase-frequency-type)
+   - ac-phase-power (ac-power-type)
+4. DC summary (dc-general )
+   - dc-current (dc-current-type)
+   - dc-voltage (dc-voltage-type)
+   - dc-power (dc-power-type)
+
+Every channel configuration consists of five parameters:
+
+1. registerFunction - identification of block where register resides.
+   - Possible are:
+     - 3 - Input register
+     - 4 - Holding register
+2. registerNumber - number of register inside of block
+3. registerType - type of register:
+   - INT
+   - INT_BIGENDIAN
+   - SHORT
+   - USHORT
+   - STATUS
+4. registerUnit - unit of value in register. Possible are all names from org.openhab.core.library.unit.Units constants. It is required to use exact names with capital Letters only. It is not verified at this moment, but when coversion fails, there is an exception thrown and message to logs is written: "Illegal access exception during reflection to Units!"
+5. registerScaleFactor - power of then that register's value should be multiplied. Eg. Daily power produced by inverter is given in Watt*Hours, and we have channel with KILO Watt*Hours, so there is need to divide register value by 1000. It means registerScaleFactor is -3.
```

Comment:
Is this parameter needed? I do not think that user should consern himself with Wh/kWh conversion as this what the whole UoM system has been huilt for. Item state description has the unit and values are converted on display.

You could consider only letting user specify unit of the data, similar to the gainOffset profile in https://github.com/openhab/openhab-addons/pull/9980: e.g. "1 mA" for milliamperes or "0.01 A" for "centi-amperes". 

Users would not need to do any unit conversions themselves, only declare the original incoming unit.

Reply:
I was designing this binding to be as generic as possible. Nobody knows what way another inverter is reporting that. 
This parameter is absolutelly needed, but possibly with other presentation to the user.
One of my colleagues has inverter reporting current power in dW, so he has to put -4 to this parameter.
I can think about how to allow users to enter this in incoming units.
Maybe in the same way as with registerUnit.
Then registerUnit should be called "presentationUnit" and this registerScaleFactor should be renamed to "incomingUnit".

Reply:
Let's continue the discussion above, it's related